### ISSUE TITLE
Not catching tabbing errors

### ISSUE THREAD

**Issue by SamMarkowitz** [CloudSlang/cloud-slang#703]
While working with @milanafinkel we ran into two issues where the syntax was a bit off, but the flow ran anyhow and therefore had unexpected behavior. Here are some simplified versions of what we ran into.

Both examples use this simple operation:

```
namespace: test_code.junk

operation:
  name: operation

  inputs:
    - to_test

  python_action:
    script: |
      print to_test
  results:
    - SUCCESS: ${to_test != 'fail'}
    - FAILURE
```

First, in this example the `on_failure` is skipped because there is no space between the `-` and the name of the step:

```
namespace: test_code.junk

flow:
  name: flow

  inputs:
    - value: 'fail'

  workflow:
    - switcher:
        do:
           operation:
             - to_test: ${value}
    - on_failure:
      -print_fail:
          do:
            print:
              - text: "failed miserably"
```

Second, in this example the `navigate` does not switch the results because it is tabbed back:

```
namespace: test_code.junk

imports:
  prop: test_code.junk

flow:
  name: flow

  inputs:
    - value: 'fail'

  workflow:
    - switcher:
        do:
          operation:
            - to_test: ${value}
      navigate:
          - SUCCESS: FAILURE
          - FAILURE: SUCCESS
```

**Comment by Bonczidai** [comment]
validation issue: related to #736

**Comment by gituser173** [comment]
second example fixed by #748
